### DISCUSSION TITLE
Select option values not honoured when using integer keys

### DISCUSSION THREAD

**Discussion by ghost** [spatie/laravel-html#177] [Ideas]
There appears to be an issue where when you're using an integer for a key the sequence gets reset to 0,1,2,3... rather than retaining the original value.

This is particularly noticeable when using `mapWithKeys` from a laravel collection i.e.

```
mapWithKeys(function ($type) {
  return [$type->id => $type->name];
})
```

This works fine when using non-int keys, however casting the int to string doesn't resolve the issue either.

Example below is used from inside a lara view component render method:

```
$label = html()->label($this->label, $this->id)->class('form-control-label');
$input = html()->select($this->name, $this->options, $this->value)
  ->id($this->id)
  ->class('form-control-input')
  ->attributes($this->attr);

$html = html()->div()->class('form-control')->addChild($label)->addChild($input);

return $html->toHtml();
```

**Comment by freekmurze** [comment]
Feel free to create a PR that fixes this. 👍
